File: Sprint1/E/C++/main.cpp
// https://contest.yandex.ru/contest/22449/problems/E/
#include <iostream>
#include <string>
#include <sstream>

int main() {
    int l;
    std::cin >> l;
    std::cin.ignore();

    std::string line;

    std::getline(std::cin, line);

    std::istringstream iss(line);
    std::string word;

    unsigned int max_len = 0;
    std::string max_len_word;

    while (iss >> word) {
        if (word.length() > max_len) {
            max_len = word.length();
            max_len_word = word;
        }
    }

    std::cout << max_len_word << std::endl;
    std::cout << max_len << std::endl;

    return 0;
}

File: Sprint1/E/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/E/

use std::io::{self, BufRead};

fn main() -> io::Result<()> {
    let stdin = io::stdin();
    let mut stdin_lock = stdin.lock();

    let mut _l = String::new();
    stdin_lock.read_line(&mut _l)?;

    let mut line = String::new();
    stdin_lock.read_line(&mut line)?;

    let (max_word, max_len) = line
        .split_whitespace()
        .map(|s| (s, s.len()))
        .max_by_key(|&(_, len)| len)
        .unwrap_or(("", 0));

    println!("{}", max_word);
    println!("{}", max_len);

    Ok(())
}
